File: client/src/app/models/workspace.ts
export type Workspace = {
    id: number;
    name: string;
    description?: string;
    createdAt: string;
    ownerId: string;
    ownerEmail?: string;
    memberCount: number;
    boardCount: number;
}

export type WorkspaceMember = {
    id: number;
    userId: string;
    email?: string;
    role: string;
}

File: API/Entities/Workspace.cs
namespace API.Entities;

public class Workspace
{
    public int Id { get; set; }
    public required string Name { get; set; }
    public string? Description { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public required string OwnerId { get; set; }
    public User Owner { get; set; } = null!;
    public ICollection<WorkspaceMember> Members { get; set; } = [];
    public ICollection<Board> Boards { get; set; } = [];
}
